db/dcrpg/pgblockchain.go

```diff
@@ -1397,31 +1397,23 @@ func (pgb *ChainDB) DevBalance() (*dbtypes.AddressBalance, error) {
 // and if cache is stale or missing data for the address, a DB query is used. A
 // successful DB query will freshen the cache.
 func (pgb *ChainDB) AddressBalance(address string) (*dbtypes.AddressBalance, error) {
-	numSpent, numUnspent, amtSpent, amtUnspent, err :=
-		pgb.AddressSpentUnspent(address)
+	balance, err := pgb.AddressSpentUnspent(address)
 	if err != nil {
 		return nil, err
 	}
-	balance := &dbtypes.AddressBalance{
-		Address:      address,
-		NumSpent:     numSpent,
-		NumUnspent:   numUnspent,
-		TotalSpent:   amtSpent,
-		TotalUnspent: amtUnspent,
-	}
 
 	return balance, nil
 }
 
 // AddressSpentUnspent attempts to retrieve balance information for a specific
 // address from cache, and if cache is stale or missing data for the address, a
 // DB query is used. A successful DB query will freshen the cache.
-func (pgb *ChainDB) AddressSpentUnspent(address string) (int64, int64, int64, int64, error) {
+func (pgb *ChainDB) AddressSpentUnspent(address string) (*dbtypes.AddressBalance, error) {
```

Comment:
The return type is changed to `dbtypes.AddressBalance` here and `RetrieveAddressSpentUnspent` to prevent adding two more return fields. Seems to make the code cleaner anyways, but maybe should rename of the `*SpentUnspent` functions.

Reply:
Seems like we should remove one of AddressBalance and AddressSpentUnspent.